src/components/Todo/Todo.js

```diff
@@ -1,70 +1,170 @@
   useEffect(() => {console.log('mount')},[]);
-  const [count, setCount] = useState(4);
-  const [items, setItems] = useState(
-    [
-      {
-        value: 'Сделать урок по React',
-        isDone: false,
-        id: 1
-      },
-      {
-        value: 'Приготовить обед',
-        isDone: false,
-        id: 2
-      },
-      {
-        value: 'Помыть посуду',
-        isDone: false,
-        id: 3
-      },
-      {
-        value: 'Погулять с собакой',
-        isDone: false,
-        id: 4
-      }
-    ]);
+  const [bottomNavigationValue, setBottomNavigationValue] = useState("all");
+  const [count, setCount] = useState(savedCount);
+  const [items, setItems] = useState(savedItems);
+
+  const moveItems = useCallback(
+    (dragIndex, hoverIndex) => {
+      const dragCard = items[dragIndex]
+      const newItemList = update(items, {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, dragCard],
+        ],
+      });
+      setItems(newItemList);
+      saveToLocalStorage(newItemList);
+    },
+    [items],
+  );
 
   const onClickDone = id => {
     const newItemList = items.map(item => {
       const newItem = { ...item };
       if (item.id === id) {
         newItem.isDone = !item.isDone;
+        if (bottomNavigationValue != "all") newItem.hide = !item.hide;
       }
       return newItem;
     });
-    setItems(newItemList);
-    };
+    setItems(newItemList); 
+    saveToLocalStorage(newItemList);
+  };
   
   const onClickDelete = id => {
     const newItemList = items.filter(item => item.id !== id);
     setItems(newItemList);
-    };
+    saveToLocalStorage(newItemList);
+  };
 
   const onClickAdd = value => {
-      setItems(
-        [...items, {
-          value,
-          isDone: false,
-          id: count + 1
-          }
-        ]);
-      setCount(count + 1)
+    let hide = (bottomNavigationValue === "completed") ? true : false;
+    const newCount=count+1;
+    const newItemList = [...items, {
+        value,
+        isDone: false,
+        id: newCount,
+        disabled: true,
+        hide: hide
+        }
+      ]
+    setItems(newItemList);
+    setCount(newCount);
+    saveToLocalStorage(newItemList);
+    localStorage.setItem("count", newCount);
   };
-  
+
+  const onEditItem = id => {
+    const newItemList = items.map(item => {
+      const newItem = { ...item };
+      if (item.id === id) {
+        newItem.disabled = false;
+      }
+      return newItem;
+    });
+    setItems(newItemList);
+    saveToLocalStorage(newItemList);
+  };
+
+  const onBlurItem = (id, newValue) => {
+    const newItemList = items.map(item => {
+      const newItem = { ...item };
+      if (item.id === id) {
+        newItem.value = newValue;
+        newItem.disabled = true;
+      }
+      return newItem;
+    });
+    setItems(newItemList);
+    saveToLocalStorage(newItemList);
+  };
+
+  const onClickDeleteCompleted = () => {
+    const newItemList = items.filter(item => item.isDone == false);
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/WebHeroSchool/agatta17_todo/pullRequest?prid=6081322)